### PR TITLE
mixin: Add Thanos Ruler alerts

### PR DESCRIPTION
**Pull request by kakkoyun**
This PR adds missing Ruler alerts to thanos-mixin.

Alerts are inspired by https://thanos.io/components/rule.md/#must-have-essential-ruler-alerts

Signed-off-by: Kemal Akkoyun <kakkoyun@gmail.com>

* [ ] I added CHANGELOG entry for this change.
* [ ] Change is not relevant to the end user.

## Changes

* Adds Ruler alerts
* Adds alert queue metric panels to Thanos Rule dashboards

## Verification

For rules:
* `make example-rules-lint`
* Manual tests using Query UI against running cluster.

For dashboards:
* Imported generated dashboard JSON to a Grafana with live data.

### CODE REVIEW

**Comment by kakkoyun** [comment]
cc @metalmatze @bwplotka

File: mixin/thanos/alerts/ruler.libsonnet [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+      {
+        name: 'thanos-ruler.rules',
+        rules: [
+          {
+            alert: 'ThanosRulerQueueIsDroppingAlerts',
+            annotations: {
+              message: 'Thanos Ruler {{$labels.job}} {{$labels.pod}} is failing to queue alerts.',
+            },
+            expr: |||
+              sum by (job) (thanos_alert_queue_alerts_dropped_total{%(selector)s}) > 0
```
Comment (metalmatze) [inline mixin/thanos/alerts/ruler.libsonnet]: `rate` is missing.

File: mixin/thanos/alerts/ruler.libsonnet [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+              severity: 'critical',
+            },
+          },
+          {
+            alert: 'ThanosRulerSenderIsFailingAlerts',
+            annotations: {
+              message: 'Thanos Ruler {{$labels.job}} {{$labels.pod}} is failing to send alerts to alertmanager.',
+            },
+            expr: |||
+              sum by (job) (thanos_alert_sender_alerts_dropped_total{%(selector)s}) > 0
```
Comment (metalmatze) [inline mixin/thanos/alerts/ruler.libsonnet]: `rate` is missing.

File: mixin/thanos/alerts/ruler.libsonnet [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+          {
+            alert: 'ThanosRulerHighRuleExaluationWarnings',
+            annotations: {
+              message: 'Thanos Ruler {{$labels.job}} {{$labels.pod}} has high number of evaluation warnings.',
+            },
+            expr: |||
+              sum by (job) (rate(thanos_rule_evaluation_with_warnings_total{%(selector)s}[5m])) > 0
+            ||| % thanos.ruler,
+
+            'for': '5m',
```
Comment (metalmatze) [inline mixin/thanos/alerts/ruler.libsonnet]: 5min seems pretty quick for this one. :thinking:

File: mixin/thanos/alerts/ruler.libsonnet [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+            alert: 'ThanosRulerRuleEvaluationLatencyHigh',
+            annotations: {
+              message: 'Thanos Ruler {{$labels.job}}/{{$labels.pod}} has higher evaluation latency than interval for {{$labels.rule_group}}.',
+            },
+            expr: |||
+              (
+                sum by (job, pod, rule_group) (prometheus_rule_group_last_duration_seconds{%(selector)s})
+              >
+                sum by (job, pod, rule_group) (prometheus_rule_group_interval_seconds{%(selector)s})
+              )
```
Comment (metalmatze) [inline mixin/thanos/alerts/ruler.libsonnet]: I'm not too sure about this rule, as it's always evaluation for us, even though everything seems fine.

File: mixin/thanos/alerts/ruler.libsonnet (hunk 3 above)
Comment (kakkoyun) [reply mixin/thanos/alerts/ruler.libsonnet]: 15m? 30m?

File: mixin/thanos/alerts/ruler.libsonnet (hunk 4 above)
Comment (kakkoyun) [reply mixin/thanos/alerts/ruler.libsonnet]: This still indicates a problem though. Its severity set to `warning`. We might also give a coefficient for the interval but I think we should have this. 

With this alert, you can detect you have too many rules for a group for example. Or your queries take too much time. 

It's not very critical but I still see value in this.

**Comment by kakkoyun** [comment]
@metalmatze ping

File: mixin/thanos/alerts/ruler.libsonnet (hunk 3 above)
Comment (kakkoyun) [reply mixin/thanos/alerts/ruler.libsonnet]: Changed to 15m.

**Comment by kakkoyun** [comment]
@metalmatze Could you merge? Or should we wait for another opinion?

**Pull request merged**